src/main/kotlin/no/nav/syfo/infrastructure/journalforing/JournalforingService.kt

```diff
@@ -0,0 +1,90 @@
+package no.nav.syfo.infrastructure.journalforing
+
+import no.nav.syfo.application.IJournalforingService
+import no.nav.syfo.domain.*
+import no.nav.syfo.infrastructure.clients.ereg.EregClient
+import no.nav.syfo.infrastructure.clients.dokarkiv.DokarkivClient
+import no.nav.syfo.infrastructure.clients.dokarkiv.dto.*
+import org.slf4j.LoggerFactory
+
+class JournalforingService(
+    private val dokarkivClient: DokarkivClient,
+    private val eregClient: EregClient,
+    private val isJournalforingRetryEnabled: Boolean,
+) : IJournalforingService {
+    override suspend fun journalfor(
+        foresporsel: Foresporsel,
+        pdf: ByteArray
+    ): Result<JournalpostId> =
+        runCatching {
+            val journalpostRequest =
+                createJournalpostRequest(
+                    foresporsel = foresporsel,
+                    pdf = pdf
+                )
+
+            val journalpostId =
+                try {
+                    dokarkivClient.journalfor(journalpostRequest).journalpostId
+                } catch (exc: Exception) {
+                    if (isJournalforingRetryEnabled) {
+                        throw exc
+                    } else {
+                        log.error("Journalføring failed, skipping retry (should only happen in dev-gcp)", exc)
+                        // Defaulting'en til DEFAULT_FAILED_JP_ID skal bare forekomme i dev-gcp:
+                        // Har dette fordi vi ellers spammer ned dokarkiv med forsøk på å journalføre
+                        // på personer som mangler aktør-id.
+                        DEFAULT_FAILED_JP_ID
+                    }
+                }
+            JournalpostId(journalpostId.toString())
+        }
+
+    private suspend fun createJournalpostRequest(
+        foresporsel: Foresporsel,
+        pdf: ByteArray,
+    ): JournalpostRequest {
+        val avsenderMottaker =
```

Comment:
Det er jo riktig at at virksomheten er mottaker her, men ble litt usikker på om sykmeldte da får se dokumentet på innloggede sider uten videre når hen bare er bruker og ikke mottaker. Må nok testes, evt sjekke opp reglene for det. Kanskje det må noe overstyring til :thinking:

Reply:
F.eks. så journalfører vi melding til behandler med behandler som mottaker og sykmeldt som bruker og gjør noe overstyring - se https://github.com/navikt/isbehandlerdialog/blob/master/src/main/kotlin/no/nav/syfo/melding/domain/MeldingTilBehandler.kt 
Kanskje det som må til her også?

Reply:
Ja, ser sånn ut. Legger til VISES_MASKINELT_GODKJENT.